### PR TITLE
Allow building and loading of TBBBind on macOS

### PR DESCRIPTION
**Pull request by isaevil**
### Description 
Current build infrastructure and code base don't allow to build and use TBBBind on macOS, while HWLOC is able to recognize hybrid cores on macOS M1/M2. With this patch user may build TBBBind on macOS and be able to use `tbb::info` namespace to obtain high-level information about the system and limit task_arena accordingly (without actual CPU binding since such functionality is not supported on macOS via HWLOC).

Fixes # - _issue number(s) if exists_

- [x] - git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/oneapi-src/oneTBB/blob/master/CONTRIBUTING.md#pull-requests) for details)_

### Type of change

_Choose one or multiple, leave empty if none of the other choices apply_

_Add a respective label(s) to PR if you have permissions_

- [ ] bug fix - _change that fixes an issue_
- [x] new feature - _change that adds functionality_
- [ ] tests - _change in tests_
- [x] infrastructure - _change in infrastructure and CI_
- [ ] documentation - _documentation update_

### Tests

- [ ] added - _required for new features and some bug fixes_
- [ ] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [x] not needed

### Breaks backward compatibility
- [ ] Yes
- [x] No
- [ ] Unknown

### Notify the following users
_List users with `@` to send notifications_

### CODE REVIEW

File: src/tbb/governor.cpp [hunk 1]
```diff
@@ -367,15 +367,18 @@ static void (*restore_affinity_ptr)( binding_handler* handler_ptr, int slot_num
...
 // Table describing how to link the handlers.
 static const dynamic_link_descriptor TbbBindLinkTable[] = {
     DLD(__TBB_internal_initialize_system_topology, initialize_system_topology_ptr),
     DLD(__TBB_internal_destroy_system_topology, destroy_system_topology_ptr),
+#if __TBB_CPUBIND_PRESENT
     DLD(__TBB_internal_allocate_binding_handler, allocate_binding_handler_ptr),
     DLD(__TBB_internal_deallocate_binding_handler, deallocate_binding_handler_ptr),
     DLD(__TBB_internal_apply_affinity, apply_affinity_ptr),
     DLD(__TBB_internal_restore_affinity, restore_affinity_ptr),
+#endif
```
Comment (ilya-lavrenov) [inline src/tbb/governor.cpp:381]: can it be dynamically understood that tbbbind does not have all symbols and "bind"-related symbols are just not set upon loading of tbbbind?

Comment (isaevil) [reply src/tbb/governor.cpp:381]: `dynamic_link` functionality guarantees that either all of requested symbols are loaded, or none of them, if something is missing. Thus, we shouldn't expect hwloc's cpubind related symbols here in the first place.

File: include/oneapi/tbb/detail/_config.h [hunk 2]
```diff
@@ -380,6 +380,9 @@
     #define __TBB_ARENA_BINDING 1
 #endif
 
+// Thread pinning is not available on macOS*
+#define __TBB_CPUBIND_PRESENT (__TBB_ARENA_BINDING && !__APPLE__)
```
Comment (pavelkumbrasev) [inline include/oneapi/tbb/detail/_config.h:384]: Does it make sense to use `!__APPLE__` directly where we need it?

File: src/tbb/arena.cpp [hunk 3]
```diff
@@ -543,10 +543,10 @@ void task_arena_impl::initialize(d1::task_arena_base& ta) {
     threading_control* thr_control = threading_control::register_public_reference();
     arena& a = arena::create(thr_control, unsigned(ta.my_max_concurrency), ta.my_num_reserved_slots, priority_level);
     ta.my_arena.store(&a, std::memory_order_release);
-#if __TBB_ARENA_BINDING
+#if __TBB_CPUBIND_PRESENT
```
Comment (pavelkumbrasev) [inline src/tbb/arena.cpp:546]: Do we really need to disable this since it will be a NOP?

**Review by pavelkumbrasev — APPROVED** [review]

**Pull request merged**